docs/self-host.md

```diff
@@ -0,0 +1,88 @@
+# Self-Hosted App
+
+<!--consumer-badges start -->
+
+![node][node-badge]
+
+<!--consumer-badges end -->
+
+## Deploy a Self-Hosted Instance
+
+Multiple options exist for deploying a self-hosted instance:
+
+### Install `@repository-settings/app` as a dependency of your own app (recommended)
+
+This option offers you the most flexibility
+
+#### Depend on the package from npm
+
+```shell
+npm install @repository-settings/app
+```
+
+#### Example node.js app
```

Comment:
in addition to this sort of example, i could see value in providing a minimal example for deploying to a serverless environment. i dont have as much experience with that approach lately, so open to suggestions. i dont think adding that needs to block getting this progress merged, but could be a valuable follow up

Reply:
I think it's a good idea, but we can do a follow up for that.